romeo/src/actor.rs

```diff
@@ -0,0 +1,171 @@
+use std::time::Duration;
+
+use futures::Future;
+use serde::{de::DeserializeOwned, Serialize};
+use tokio::{
+    sync::broadcast::{self, error::RecvError, Sender},
+    task::JoinHandle,
+    time::sleep,
+};
+
+use crate::event::Event;
+use crate::store::Store;
+
+pub trait Actor: Serialize + DeserializeOwned + Send + Sync + 'static {
+    const NAME: &'static str;
+
+    fn handle(&mut self, event: Event) -> anyhow::Result<Vec<Event>>;
```

Comment:
Hmm yeah I like this approach. Right now we have designed this application to be composed of many small actors in which case they will contain fallible functions, but it might more sense to just model the application as one big state whose state transitions is defined purely by a series of events.

@stjepangolemac let's think about this a bit tomorrow.

Reply:
I'm thinking something along these lines. The whole app would essentially be encapsulated in a "State" struct and an "update" function.
![image](https://github.com/stacks-network/sbtc/assets/22524681/21b892b2-d926-4fe2-bf09-35b1458f590d)

Reply:
Yes, and it's very easy to mock input for testing different scenarios and property testing.

Reply:
BTW, if we would like to use multiple threads and no to block I/O by unrelated tasks, we can have a set of isolated synchronous actors/state_machines, and call them in a thread pool. In this case, each state machine can produce other state machines. The system then will add them to a queue. Each state machine must be completely isolated from I/O and should not communicate between each other directly.

Reply:
Can we simplify this further and say that tasks are just additional types of Events?

That would make it much closer to what we have now. Also, I kinda like the idea of just one large actor / state where we cover all the events. Having multiple actors might make the logic clearer, but starting with one might help us not to overstep in decoupling everything.

Reply:
Yeah what I imagined in the picture is that tasks are processes that do IO and spawned asynchronously. They emit events, which we use to update the application state.

The simplification here is that tasks are the only entities which create events, and the state update logic can only update the internal state and return new tasks to be spawned.

Reply:
We have changed the structure now according to this discussion. It feels much simpler now so I think we're in a good position to move forward with this.